### ISSUE TITLE
Time not properly encoded in wb-feed plugin

### ISSUE THREAD

**Issue by goatsweater** [wet-boew/wet-boew#8605]
The current implementation of the wb-feed plugin writes dates/times in a format that is not entirely conformant to expectations from screen readers, effectively rendering dates unparsable by VoiceOver.

As implemented, dates/times get rendered as `<time>2019-03-07 15:52</time>`, with the parent element having a `.feeds-date` class that wraps the `<time>` in square brackets using ::before and ::after content. This makes VoiceOver read only the bracket and not understand the date.

Recommended solution:
Dates/times should be encoded as shown at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time, with the `datetime` attribute added to the `<time>` element. This will allow machine readable dates and times to be parsed correctly by assistive technologies.

**Comment by EricDunsworth** [comment]
> This makes VoiceOver read only the bracket and not understand the date.

@goatsweater Do you mean it's only announcing the opening and closing brackets? What is it currently announcing? What would you expect it to announce?

> Recommended solution:
Dates/times should be encoded as shown at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time, with the ``datetime`` attribute added to the <time> element. This will allow machine readable dates and times to be parsed correctly by assistive technologies.

Do you know for a fact that this would make a difference to VoiceOver in this context?

The dates/times on that page are already written as [valid local date and time strings](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-time-element:valid-local-date-and-time-string) (which is a misrepresentation of what's in the feeds..). The HTML spec claims that the ``datetime`` value can be derived from either the namesake attribute or from the [content that's in-between the ``<time></time>`` tags](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-time-element:child-text-content). So adding a ``datetime`` attribute with the same content would be redundant as far as the spec is concerned.

Just tried a test in NVDA 2019.2.1 + Firefox Developer Edition 74.0b1 on the [feeds demo page](https://wet-boew.github.io/v4.0-ci/demos/feeds/feeds-en.html)...

* By default it announced the first ATOM feed's date (``<small class="feeds-date"><time>2018-10-19 12:06</time></small>``) as "Two thousand and eighteen ten nineteen twelve o six". It didn't mention the brackets.
* Nothing changed after I added a ``datetime="2018-10-19 12:06"`` attribute to the ``<time>`` element.

**Comment by goatsweater** [comment]
It seems like this is fixed with the change in theme on canada.ca. I went to several places where content with dates are produced through feeds and tried them - all read the date consistently well regardless of the format it was shown on the page.

Also, there are no longer any brackets being displayed around dates, which I think was the cause of the problem originally.

**Comment by goatsweater** [comment]
The demo page does work. I tried it on both MacOS and iOS, both using the built in accessibility screen reader. I'm guessing they are the same engine under the hood, but both ignore the brackets completely. I'm using MacOS 10.14.6 and iOS 13.3.1, and I used Safari to load the page and have it read to me.

It is reading the page like a normal person would announce the date now. It reads as October 19, 2018, on the first example feed entry and properly pauses between the text and the date. Previously it was reading dates as "bracket two thousand eighteen dash ten dash 19" as if it was just part of the text.

**Comment by EricDunsworth** [comment]
Awesome :)! My guess is that it was that probably a bug in older versions of VoiceOver that has since been addressed (since the HTML semantics were valid... apart from misrepresenting the time due to #8840).
